corehq/apps/app_manager/xform.py

```diff
@@ -1567,12 +1567,13 @@ def _create_casexml_2(self, form):
         if form.requires == 'none' and not form_opens_case and 'update_case' in actions:
             raise CaseError("To update a case you must either open a case or require a case to begin with")
 
+        module = form.get_module()
+        is_registry_case = _module_offers_registry_search(module) and not form_opens_case
         delegation_case_block = None
         if not actions or (form.requires == 'none' and not form_opens_case):
             case_block = None
         else:
-            case_block = XFormCaseBlock(self)
-            module = form.get_module()
+            case_block = XFormCaseBlock(self) if not is_registry_case else None
```

Comment:
Won't setting `case_block` to `None` potentially fail later on, like the `case_block.add_create_block` in the `if form_opens_case` block below?

Reply:
I think not as it's currently written, though the logic here is kinda hard to follow.  `case_block` is always defined if `form_opens_case = True`.

Reply:
ohhh, I get it now - thanks!

Reply:
I agree that this approach isn't great. I've made a change to make the XFormCaseBlock lazy: 3a1f087e2ac

Reply:
Thank you, it's easier to follow now.

Reply:
seems that's broken a bunch of tests

Reply:
reverted that change since it seems it's more effort than it's worth

Reply:
I took another pass: https://github.com/dimagi/commcare-hq/pull/30532